core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -134,125 +152,129 @@ public boolean isEmpty() {
     }
 
     @Override
-    public CompletableFuture<Void> completionFuture() {
-        return completionFuture;
+    long demand() {
+        return pendingDemand;
     }
 
     @Override
-    public void subscribe(Subscriber<? super T> subscriber) {
-        requireNonNull(subscriber, "subscriber");
-        subscribe0(subscriber, null, false);
+    void request(long n) {
+        Subscription subscription = delegateSubscription;
+        if (subscription != null) {
+            subscription.request(n);
```

Comment:
nit: Could we add before here `assert pendingDemand == 0;` just to confirm that there is no pending demand?

Reply:
BTW, I ended up having to make the ForwardingSubscriber's onSubscribe slightly more complex to satisfy the assertion. Not a big deal, but before the pendingDemand usage was still guaranteed to be correct with a bit less complexity. Let me know if you think it's better this way.

Reply:
Oh.. I think removing the assertion is okay for less complexity.

Reply:
Thanks :)